nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/pause/StandardRecordProcessorBlocker.java

```diff
@@ -38,7 +76,17 @@ public synchronized void block() {
                 : new CountDownLatch(1);
     }
 
-    public synchronized void unblock() {
+    public void unblock() {
+        unblockPrivate(timestampProvider.get());
+    }
+
+    public void unblockAndDisableTimeout() {
+        unblockPrivate(Long.MAX_VALUE);
+    }
+
+    private synchronized void unblockPrivate(final long newTimeoutAnchor) {
```

Comment:
Use of `Private` in the method name should be avoided. `Internal` could work, but better to be more descriptive, such as `countDown()` based on what the method is doing.